### ISSUE TITLE
Describe ScalaDoc syntax and semantics

### ISSUE THREAD

**Issue by Toxaris** [scala/scala.github.com#526]
Yesterday, I was using ScalaDoc for real for the first time, and I had to figure some aspects of the syntax and semantics out by experiments or looking at random Scala sources. @Blaisorblade convinced me I should report the missing bits of information somewhere so they can be fixed :) I hope this is the right place, please direct me to the right place if not.

To learn about ScalaDoc, I started reading http://docs.scala-lang.org/style/scaladoc.html and then followed the link to https://wiki.scala-lang.org/display/SW/Writing+Documentation. Maybe there is something else I should have read?

I remember having trouble to figure the following bits out:

- to link to companion objects, use a `$` at the right place
- to use your own link text, put after the link target in the double brackets
- bullet lists can nest
- the scope of `@define` is to the end of the doc comment or to the next `@define`
- macros are inherited from super classes to objects, but docs are not

I'm also confused about how to best document a hierarchy of case classes with some intermediate traits. I would like for readers to look at the whole hierarchy at once, not at each case class separately. But I also don't want to repeat the hierarchy manually because I plan to extend it and I'm afraid I'll not keep the docs up-to-date. The style guide tells me to document each case class separately, but nothing about how to document the whole hierarchy.

**Comment by SethTisue** [comment]
@VladUreche there aren't other authoring docs @Toxaris missed, are there?

**Comment by VladUreche** [comment]
I guess this old presentation may help as well: http://www.slideshare.net/VladUreche/scaladoc-reflection. I also gave some notes I had to @janekdb hoping he will publish them, but they are very crappy.

I think someone should make a push to document Scaladoc, since it has a lot of nice features, but I just don't have the time to take care of this. On the other hand, I'd be glad to help anyone who steps up by answering any question they have.

**Comment by janekdb** [comment]
I have been making glacial progress on https://github.com/scala/scala.github.com/pull/521 and have the links from Vlad to use. I'll have a go at picking up the pace a bit.

**Comment by SethTisue** [comment]
another old ticket: https://github.com/scala/scala-lang/pull/394

**Comment by dragos** [comment]
Yeah, this is quite annoying, I'll give it a try. I often need to google for that forgotten wiki page. Where should this go, docs.scala-lang or scala-lang.org? I'm guessing the former.

**Comment by heathermiller** [comment]
That would be awesome, thanks Iuli. It should go to docs.scala-lang

**Comment by janekdb** [comment]
Shamefacedly I have to reveal that I redirected my energies into Chapter 8 of https://www.amazon.co.uk/Professional-Scala-Janek-Bogucki/dp/1119267226 toward the end of last year.

Don't be deceived by the one-star review - over the course of 45 pages everything I could find about Scaladoc is surveyed and explained including quite a lot of hitherto undocumented capability. I believe I consulted the forgotten wiki page when putting this together.

At the least it could be used as a checklist for future documentation efforts.

**Comment by janekdb** [comment]
Shamefacedly I have to reveal that I redirected my energies into Chapter 8
of https://www.amazon.co.uk/Professional-Scala-Janek-Bogucki/dp/1119267226
toward the end of last year.

Don't be deceived by the one-star review - over the course of 45 pages
everything I could find about Scaladoc is surveyed and explained including
quite a lot of hitherto undocumented capability. I believe I consulted the
forgotten wiki page when putting this together.

At the least it could be used as a checklist for future documentation
efforts.

**Comment by SethTisue** [comment]
it sounds like there are multiple sources of documentation and it's easy to find one and think it's all there is, when actually there's more.

ideally maybe there'd be some big merge, but perhaps in the meantime someone would like to do a more modest PR (or group of PRs, if multiple repos are involved) that just adds cross-links? (including Janek's thing, I don't see a reason not to link to that too)

**Comment by xolve** [comment]
There are two important pages [Scaladoc Style Guide](https://docs.scala-lang.org/style/scaladoc.html) and [Scaladoc For Library Authors](https://docs.scala-lang.org/overviews/scaladoc/for-library-authors.html). These seem like quick tutorials than reference documents. e.g. While browsing Scale code I see sytnax which I cannot find.

[Scaladoc Style Guide](https://docs.scala-lang.org/style/scaladoc.html) suggests `Use the wiki-style syntax instead of HTML wherever possible.`. What wiki-style means here is not clear.

**Comment by laughedelic** [comment]
@xolve you can find that wiki-syntax in the second document in the [Markup](https://docs.scala-lang.org/overviews/scaladoc/for-library-authors.html#markup) section.

**Comment by xolve** [comment]
Thank you @laughedelic this is very helpful 👍

**Comment by SethTisue** [comment]
another piece of doc: https://www.scala-lang.org/blog/2018/10/04/scaladoc-tables.html

**Comment by SethTisue** [comment]
I'm going to close this ticket, since otherwise it's likely to stay open forever

* main Scaladoc 3 doc is https://docs.scala-lang.org/scala3/guides/scaladoc/
* main Scaladoc 2 doc is:
  * https://docs.scala-lang.org/overviews/scaladoc/for-library-authors.html
  * https://docs.scala-lang.org/style/scaladoc.html

doubtless both of them could be improved further
